x-pack/plugins/maps/server/maps_telemetry/maps_telemetry.ts

```diff
@@ -82,6 +94,112 @@ function getIndexPatternsWithGeoFieldCount(indexPatterns: IIndexPattern[]) {
   };
 }
 
+function getEMSLayerCount(layerLists: LayerDescriptor[][]): ILayerTypeCount[] {
+  return layerLists.map((layerList: LayerDescriptor[]) => {
+    const countsById = _(layerList).countBy((layer: LayerDescriptor) => {
+      const isEmsFile =
+        layer.sourceDescriptor !== null
+          ? layer.sourceDescriptor.type === SOURCE_TYPES.EMS_FILE
+          : false;
+      const id = (layer.sourceDescriptor as AbstractSourceDescriptor).id;
+      return isEmsFile && id ? id : 'false';
+    });
+
+    const emsCountsById = countsById.pickBy((val, key) => key !== 'false');
+    const layerTypeCount = emsCountsById.value();
+    return layerTypeCount as ILayerTypeCount;
+  }) as ILayerTypeCount[];
+}
+
+function isGeoshapeIndexPattern(
```

Comment:
Logic of passing in indexPatterns is flawed and you may not find the target index pattern when there are more then 20 since saved object client will only return 20 without paging. https://github.com/elastic/kibana/issues/51345

This change is not needed in this PR but telemetry needs to be updated to avoid this problem. Instead, just used the saved object client to find the index pattern by id. Then cache that index pattern so if its needed again, it can be accessed instead of re-fetched.

Reply:
+1 fix separately, known issue that affects a few other metrics as well